### PR TITLE
Implement new ML model that has been trained on all available items

### PR DESCRIPTION
**Pull request by cainy393**
# Checklist:

Related work items: #261 

Preview url: https://hsk-training.firebaseapp.com

- [ ] Tests have been added for new functionality
- [x] Work has been reviewed by other team members
- [ ] Work has been reviewed by @craigayre or @eclennett

### CODE REVIEW

File: src/components/ItemRecognition/ItemRecognition.js [hunk 1]
```diff
@@ -75,19 +78,40 @@ class ItemRecognition extends Component {
     return (Date.now() - this.scanningStartTime) / 1000 > seconds;
   }
 
+  isItemRecognised = items => {
+    this.predictionStack.push(items.slice(0, ML_CERTAINTY));
+    const predictionCount = this.predictionStack.length;
+    if (predictionCount < ML_CERTAINTY) return false;
+    for (let i = 1; i < ML_CERTAINTY; i++) {
```
Comment (craigayre) [inline src/components/ItemRecognition/ItemRecognition.js]: Found this a bit hard to read initially, would inverting the logic and extracting to some booleans help?

How does something like this read in comparison?

```
isItemRecognised = items => {
  this.predictionStack.push(items.slice(0, ML_CERTAINTY))

  if (this.predictionsStack.length < ML_CERTAINTY) {
    return false
  }

  const predictionRatioThreshold = 1.2

  const predictions = this.predictionStack.slice(-ML_CERTAINTY) // Get only the number we care about. Needs a better name as it's really of the form [[prediction, prediction2, prediction3, ...], [...]] 
  
  const { id } = predictions[0][0]
  const areAllBestPredictionsTheSame = predictions.slice(1).every(([bestPrediction]) => bestPrediction.id === id)

  const areAllBestPredictionsCertainEnough = predictions.every(([bestPrediction, secondBestPrediction]) => bestPrediction.value / secondBestPrediction.value > predictionRatioThreshold)

  return areAlllBestPredictionsTheSame && areAllBestPredictionsCertainEnough
```

What do you think?

File: src/components/ItemRecognition/ItemRecognition.js [hunk 2]
```diff
@@ -31,8 +31,11 @@ class ItemRecognition extends Component {
     modelLoaded: false
   };
 
+  predictionStack = [];
+
   componentDidMount() {
     this.props.setPrediction(null, null);
+    this.predictionStack = [];
```
Comment (craigayre) [inline src/components/ItemRecognition/ItemRecognition.js]: Is this needed in `componentDidMount` as well?

File: src/components/ItemRecognition/ItemRecognition.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ import MobileNet from '../Admin/Trainer/MobileNet';
 import './ItemRecognition.css';
 
 const TIMEOUT_IN_SECONDS = 10;
-const ML_THRESHOLD = 0.35;
+const ML_CERTAINTY = 3;
```
Comment (craigayre) [inline src/components/ItemRecognition/ItemRecognition.js]: Is there any way we could make this easier to know exactly what it's used for at a glance?

It's almost how many of the same guesses do we need to be certain

Not sure what's the best for this. Might be good to ask another member of the team, explain what it's used for and that'll show us a good descriptive name for it

Comment (HelloUniverseNo31415) [reply src/components/ItemRecognition/ItemRecognition.js]: MIN_CONSECUTIVE_GUESSES/MATCHES or something?

File: src/components/ItemRecognition/ItemRecognition.js (hunk 2 above)
Comment (cainy393) [reply src/components/ItemRecognition/ItemRecognition.js]: Wanted to make sure it was empty at the start of each use. Only defined it outside of componentDidMount to make sure it existed the first time.

File: src/components/ItemRecognition/ItemRecognition.js (hunk 1 above)
Comment (cainy393) [reply src/components/ItemRecognition/ItemRecognition.js]: Much easier to follow, thanks.

**Pull request merged**